flink-connector-jdbc-core/src/main/java/org/apache/flink/connector/jdbc/core/util/VisibleForTest.java

```diff
@@ -0,0 +1,20 @@
+package org.apache.flink.connector.jdbc.core.util;
+
+import org.apache.flink.annotation.PublicEvolving;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Target;
+
+/**
+ * This annotations declares that a function, field, constructor, or entire type, is only visible
+ * for testing purposes.
+ *
+ * <p>This annotation is typically attached when for example a method should be {@code private}
+ * (because it is not intended to be called externally), but cannot be declared private, because
+ * some tests need to have access to it.
+ */
+@Documented
+@Target({ElementType.TYPE, ElementType.METHOD, ElementType.FIELD, ElementType.CONSTRUCTOR})
+@PublicEvolving
+public @interface VisibleForTest {}
```

Comment:
I'm curious why it is necessary to create this new class?

I looked at the VisibleForTesting class in Flink and found that it is also annotated as `Internal`.

Is the reason for introducing this class the same as the purpose of `Preconditions`?
If so, if we decide to use this approach, we need to modify the relevant configuration items in `checkstyle.xml` and add the likely comments like the new `Preconditions` to clarify why need it.
Alternatively, can we initiate a discussion to change the annotations of these classes (VisibleForTesting and Preconditions) in the Flink package to `non-internal`? (I think there’s probably missed some context in historical discussion about the annotations for related classes from my side. pls correct me if I'm wrong. )

Thanks!

Reply:
We are duplicating the class because in Flink is annotated with internal and architecture tests fail because of that.. we cannot use any class annotated with internal (this is the same reason for Preconditions and we could have more classes with this problem)

I could revert this changes to allow the discussion, I also think that change on Flink class both classes to non-internal would be great.

Reply:
Maybe there’s no rush to modify anything before we reaching a conclusion.

I'm not familiar with the entire testing framework and the use of annotations in Flink, so I couldn't make useful ideas.
If we decide to change the relevant annotations and utility classes to `non-internal`, we should confirm what impact this will have on the main Flink repository. If the impact is minimal, maybe we can initiate a discussion.

Hi, @MartijnVisser could you help give some ideas about it?